### DISCUSSION TITLE
External (non local) API address

### DISCUSSION THREAD

**Discussion by dazzakiller** [coddingtonbear/obsidian-web#34] [Ideas]
Hi @coddingtonbear ,

First thanks for all your work in producing these plugins, they have and are helping me and streamlining my work flows.

I'm hosting Obsidian on a different machine from my local, in a docker container (on a server) so I can have access through a browser. I've opened and directed both ports 27213/4 to my server running my Obsidian instance. 

When I change the API URL while clipping a webpage, the plugin is still pretending the URL with the local IP (127.0.01). Thus I'm getting the following error below. 
Is there a way to add a switch or config field for the base URL?

Error
Could not send content to Obsidian: TypeError: Failed to execute 'fetch' on 'Window': Failed to parse URL from http://127.0.0.1:27123https://xxxxxxxx.xxx/vault/· Encounters/Inbox/Inbox.md

**Comment by coddingtonbear** [comment]
I'm afraid that for security reasons, the way both the obsidian-local-rest-api plugin and this tool work prevents you from accomplishing that in the way you're trying to do it.  Good news, though -- there's an easy workaround: just use ssh port forwarding.

I don't store my Obsidian notes on my work laptop, but still would like to be able to use obsidian-web for sending pages to my notes; so what I do is just run Obsidian on my personal laptop (which will dutifully listen on port 27123 and 27124 for incoming REST requests), and use ssh port forwarding for forwarding requests to my work laptop's port 27123 to my personal laptop.

The particular setup I have in my `.ssh/config` is this:

```
Host my-personal-laptop
HostName my-personal-laptop-hostname.local
LocalForward 127.0.0.1:27124 127.0.0.1:27124
LocalForward 127.0.0.1:27123 127.0.0.1:27123
```

After you set up a similar config (of course, replacing 'my-personal-laptop-hostname' and whatnot with the hostname of the computer running Obsidian), you'll be able to run `ssh my-personal-laptop` and your browser's requests will be tunnelled over to the correct computer.

Cheers & good luck!

**Reply by dazzakiller** [reply]
Hi @coddingtonbear,

Thanks so much for your reply; that is helpful. I've not managed to get it working just yet, I think because I'm doing this through docker, I'm getting a few issues that I need to resolve. But this is very helpful and has pointed me in a direction that I hope will lead to success.

Again thank you.

**Reply by shanness** [reply]
Just for anyone considering this, consider autossh  just get it working then swap your command from ssh <blah> to autossh <blah> and it'll stay up.  Put in a startup script, and you've got a pretty bulletproof tunnel :)

**Comment by coddingtonbear** [comment]
Incidentally -- as of v3.0.0 you can now configure Obsidian Web to connect to any host.
